guvnor-project/guvnor-project-backend/src/test/java/org/guvnor/common/services/project/backend/server/ProjectRepositoriesServiceTest.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.common.services.project.backend.server;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.guvnor.common.services.backend.util.CommentedOptionFactory;
+import org.guvnor.common.services.project.model.MavenRepositoryMetadata;
+import org.guvnor.common.services.project.model.MavenRepositorySource;
+import org.guvnor.common.services.project.model.Project;
+import org.guvnor.common.services.project.model.ProjectRepositories;
+import org.guvnor.common.services.project.service.ProjectRepositoriesService;
+import org.guvnor.common.services.project.service.ProjectRepositoryResolver;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.uberfire.backend.server.util.Paths;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.io.IOService;
+import org.uberfire.java.nio.base.options.CommentedOption;
+import org.uberfire.java.nio.file.FileSystem;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ProjectRepositoriesServiceTest {
+
+    @Mock
+    private IOService ioService;
+
+    @Mock
+    private ProjectRepositoryResolver<Project> repositoryResolver;
+
+    @Mock
+    private CommentedOptionFactory commentedOptionFactory;
+
+    private ProjectRepositoriesService service;
+    private ProjectRepositoriesContentHandler contentHandler;
+
+    @Before
+    public void setup() {
+        contentHandler = spy( new ProjectRepositoriesContentHandler() );
+        service = new AbstractProjectRepositoriesService<Project>( ioService,
+                                                               repositoryResolver,
+                                                               contentHandler,
+                                                               commentedOptionFactory ) {
+        };
+    }
+
```

Comment:
@Rikkola ```MavenProjectServiceImpl``` is in ```guvnor-webapp``` which does not support checking whether a GAV already exists - well, it could have support, but it's not implemented hence the ```DeploymentMode``` parameter is not used.